processing/src/main/java/org/apache/druid/segment/IndexIO.java

```diff
@@ -449,59 +455,71 @@ public QueryableIndex load(File inDir, ObjectMapper mapper) throws IOException
         if (index.getSpatialIndexes().get(dimension) != null) {
           builder.setSpatialIndex(new SpatialIndexColumnPartSupplier(index.getSpatialIndexes().get(dimension)));
         }
-        columns.put(
-            dimension,
-            builder.build()
-        );
+        if (lazy) {
+          columns.put(dimension, Suppliers.memoize(() -> builder.build()));
+        } else {
+          ColumnHolder columnHolder = builder.build();
+          columns.put(dimension, () -> columnHolder);
+        }
```

Comment:
this pattern of code is repeated in many places, is it possible to add a private method `Supplier<ColumnHolder> getColumnHolderSupplier(ColumnBuilder builder, boolean lazy)` and use that here and all other places like

```suggestion
        columns.put(dimension, getColumnHolderSupplier(builder, lazy);
```

Reply:
done